### ISSUE TITLE
XP issues

### ISSUE THREAD

**Issue by DisposableKatyusha** [deferred-impact/battledudes-issue-tracker#39]
I do recall d21 has stated a problem with them, and that is that they're confusing when stating killstreaks. (I do not remember it but ok, here's teh link: https://github.com/deferred-impact/battledudes-issue-tracker/issues/17)

More things that have to be added:
- State everything you do with the amount of XP you gain. (i.e: Objective Capture + (number) XP) Purpose is to let people know that playing the objective is crucial for yourself too, not just getting kills.
- More XP types:
- Destroying enemy equipment/explosives
- Healing
- Resupplying
- Repairing
- Assist (originally suggested by haox)
- Assist counts as kill (when you shoot a guy until he has barely any health left, and someone else finishes him off, both of you get the kill. It must be done in the time window where he doesn't regenerate health.)
- **Destroying a vehicle/destroying vehicle assist XP (Idea trashed)**

**Comment by deferred-impact** [comment]
Obligatory "no XP for destroying empty cars wtf guys" from me. Otherwise seems reasonable. Assist XP should be discussed in the other ticket

**Comment by DisposableKatyusha** [comment]
I was curious about the last one because usually Battlefield does that sort of system; though admittedly their vehicle spawning system (at least from Battlefield 4 onwards) is much different than here, usually it's from the spawn menu and *not* in the field. I recall Battlefield 3 had vehicles in the field that you could use but I do not know if there was any problems with people destroying cars there.

**Comment by deferred-impact** [comment]
Even if they didn't do it in Battlefield, keep in mind that it's a "big game" thingy which costs a lot of money and can only be played by first installing it, creating large barriers to entry which mostly filter out idiots.

This is an .io game, full of idiots. If you think they aren't stupid enough to just go around blowing up cars for XP, you are high and/or crazy. 

Especially if you consider that C4 respawn glitch. What stops me from
 - choosing a C4 in my loadout
 - bringing a huge amount of cars into an area
 - sticking a single C4 on each of them
 - destroying them over and over, getting XP, and having them respawn near me because of the C4

**Comment by DisposableKatyusha** [comment]
Ok, point taken. So for now, anything to do with vehicles will probably be thrown out.

**Comment by Gregory-Y** [comment]
I agree, maybe I'll come up with some actual values later and post them here.

**Comment by DisposableKatyusha** [comment]
So for the most part, three people (nlpha, d21, NOTGregory) agree overall that this list (by no means exhaustive) should at least be added. Destroying vehicles have been trashed. Consensus it is.

**Comment by Gregory-Y** [comment]
Yep consesus it is, at least on the general idea of the XP issues.

Here's my idea regarding how XP should be earned, feel free to elaborate upon it.
Destroying enemy equipment/explosives: 10XP.
Healing: 1-2 XP for every heal (every time someone's HP increases).
Resupplying: 1-2 XP for every time someone gains ammo or a throwable.
Repairing: 1XP for every hit from a repair gun that repairs an allied vehicle with someone already in it.

Assist: Earn the same XP as the number of damage that you dealt to the player (provided that it was more than 10; e.g. you deal 28.5 damage to someone and earn 29 XP)

Assist counts as kill: When you shoot a guy until he has barely any health left, and someone else finishes him off, both of you get the kill, but you don't earn any XP because you stole it.

**Comment by deferred-impact** [comment]
I fear those numbers are too high and will encourage people to farm with ammo boxes & medkits & repair guns

I.e. why spend effort to get 4 kills if you could use up a full repair gun?

**Comment by DisposableKatyusha** [comment]
Based on this numbers given doesn't 4 kills give more XP immediately? It is just enough such that it rewards the player for helping his team while just small enough such that it doesn't go overboard. Also, a guy who resupplies and heals everyone is not really a problem if that's what you're asking.

**Comment by Gregory-Y** [comment]
Yeah and we could make it so it doesn't give you xp if you resupply yourself.

**Comment by deferred-impact** [comment]
Objective XP should be doubled if your team is behind on points
